Sources/MapboxMaps/Annotations/AnnotationOrchestratorImpl.swift

```diff
@@ -0,0 +1,262 @@
+import Foundation
+import UIKit
+@_implementationOnly import MapboxCommon_Private
+
+internal protocol AnnotationOrchestratorImplProtocol: AnyObject {
+    var annotationManagersById: [String: AnnotationManager] { get }
+    func makePointAnnotationManager(id: String,
+                                    layerPosition: LayerPosition?,
+                                    clusterOptions: ClusterOptions?) -> PointAnnotationManager
+    func makePolygonAnnotationManager(id: String, layerPosition: LayerPosition?) -> PolygonAnnotationManager
+    func makePolylineAnnotationManager(id: String, layerPosition: LayerPosition?) -> PolylineAnnotationManager
+    func makeCircleAnnotationManager(id: String, layerPosition: LayerPosition?) -> CircleAnnotationManager
+    func removeAnnotationManager(withId id: String)
+}
+
+internal final class AnnotationOrchestratorImpl: AnnotationOrchestratorImplProtocol {
+
+    private let tapGestureRecognizer: UIGestureRecognizer
+
+    private let longPressGestureRecognizer: MapboxLongPressGestureRecognizer
+
+    private let style: StyleProtocol
+
+    private let mapFeatureQueryable: MapFeatureQueryable
+
+    private let offsetPointCalculator: OffsetPointCalculator
+
+    private let offsetLineStringCalculator: OffsetLineStringCalculator
+
+    private let offsetPolygonCalculator: OffsetPolygonCalculator
+
+    private weak var displayLinkCoordinator: DisplayLinkCoordinator?
+
+    internal init(tapGestureRecognizer: UIGestureRecognizer,
+                  longPressGestureRecognizer: MapboxLongPressGestureRecognizer,
+                  mapFeatureQueryable: MapFeatureQueryable,
+                  style: StyleProtocol,
+                  displayLinkCoordinator: DisplayLinkCoordinator,
+                  offsetPointCalculator: OffsetPointCalculator,
+                  offsetLineStringCalculator: OffsetLineStringCalculator,
+                  offsetPolygonCalculator: OffsetPolygonCalculator) {
+        self.tapGestureRecognizer = tapGestureRecognizer
+        self.longPressGestureRecognizer = longPressGestureRecognizer
+        self.mapFeatureQueryable = mapFeatureQueryable
+        self.style = style
+        self.displayLinkCoordinator = displayLinkCoordinator
+        self.offsetPointCalculator = offsetPointCalculator
+        self.offsetLineStringCalculator = offsetLineStringCalculator
+        self.offsetPolygonCalculator = offsetPolygonCalculator
+
+        tapGestureRecognizer.addTarget(self, action: #selector(handleTap(_:)))
+        longPressGestureRecognizer.addTarget(self, action: #selector(handleDrag(_:)))
+    }
+
+    /// Dictionary of annotation managers keyed by their identifiers.
+    internal var annotationManagersById: [String: AnnotationManager] {
+        annotationManagersByIdInternal
+    }
+
+    private var annotationManagersByIdInternal = [String: AnnotationManagerInternal]()
```

Comment:
Why do we have duplication here? Can it be replaced with an internal getter and a private setter?

Reply:
There is slight difference here between the two - privately the implementation stores managers as `AnnotationManagerInternal`(to be able to utilize internal methods) and it exposes them outside as `AnnotationManager`.